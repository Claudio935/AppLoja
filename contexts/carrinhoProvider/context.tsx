import React,{ createContext } from 'react'
import { Action, PropsCarrinho, PropsCarrinhoState } from '../../types/interfaces';









export const defaultValue: PropsCarrinhoState =  {
  carrinho:[
  ]}


export const CarrinhoContext = createContext<PropsCarrinhoState>(defaultValue)