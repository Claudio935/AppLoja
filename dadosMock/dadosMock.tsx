export const mockCarrinho = [{
    title: 'testMasculino',
    image: './img/testMasculino',
    price: 1.5,
    description: 'testandoMasculino',
    id: '1',
    category: "men's clothing"
  
  },
  {
    title: 'testFeminino',
    image: './img/testFeminino',
    price: 2,
    description: 'testandoFeminino',
    id: '2',
    category: "women's clothing"
  
  },
  {
    title: 'testEletronico',
    image: './img/testEletronico',
    price: 3,
    description: 'testandoEletronico',
    id: '3',
    category: "electronics"
  
  },
  
  {
    title: 'testJoia',
    image: './img/testJoia',
    price: 4,
    description: 'testandoJoia',
    id: '4',
    category: "jewelery"
  
  }
  ]