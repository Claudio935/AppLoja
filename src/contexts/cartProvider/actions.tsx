import { PropsCart, type Action, } from "../../types/types"
import { types } from "./types"

export const addProduto = (dispatch: React.Dispatch<Action>, payload: PropsCart): void => {
  dispatch({ type: types.TYPE_ADD_CART, payload })
}

export const deleteProduct = (dispatch: React.Dispatch<Action>, payload: PropsCart): void => {
  dispatch({ type: types.TYPE_DELETE_CART, payload })
}
export const IncrementProduct = (dispatch: React.Dispatch<Action>, payload: PropsCart): void => {
  dispatch({ type: types.TYPE_INCREMENT_CART, payload })
}
export const decrementProduct = (dispatch: React.Dispatch<Action>, payload: PropsCart): void => {
  dispatch({ type: types.TYPE_DECREMENT_CART, payload })
}

