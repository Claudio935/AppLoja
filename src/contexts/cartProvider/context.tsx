import { createContext } from 'react'
import { type Action, type PropsCartState } from '../../types/types';


export const defaultValue: PropsCartState = {
  cart: []
}


export const CartContext = createContext<{
  store: PropsCartState;
  dispatch: React.Dispatch<Action>;
}>({
  store: defaultValue,
  dispatch: () => null
})