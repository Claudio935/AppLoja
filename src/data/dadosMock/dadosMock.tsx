export const mockCart = {
  cart: [{
    title: 'testMasculino',
    image: './img/testMasculino',
    price: 1.5,
    description: 'testandoMasculino',
    id: '1',
    category: "men's clothing",
    quantidade: 1,
  },
  {
    title: 'testFeminino',
    image: './img/testFeminino',
    price: 2,
    description: 'testandoFeminino',
    id: '2',
    category: "women's clothing",
    quantidade: 1,
  },
  {
    title: 'testEletronico',
    image: './img/testEletronico',
    price: 3,
    description: 'testandoEletronico',
    id: '3',
    category: "electronics",
    quantidade: 1,
  },

  {
    title: 'testJoia',
    image: './img/testJoia',
    price: 4,
    description: 'testandoJoia',
    id: '4',
    category: "jewelery",
    quantidade: 1,

  }
  ],
  dispatch: () => null
}

export const mockUseContext = {
  store: {
    cart:
      [
        {
          title: 'testMasculino',
          image: './img/testMasculino',
          price: 1.5,
          description: 'testandoMasculino',
          id: '102',
          category: "men's clothing",
          quantidade: 1,
        },
      ],
  },

  dispatch: jest.fn()
}