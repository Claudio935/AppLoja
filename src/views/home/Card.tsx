import React from 'react';
import {
  Image,
  Text,
  View,
  TouchableOpacity,
  StyleSheet
} from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { type PropsCart } from '../../types/types';
import { type StackNavigation } from '../../../App';

import { transformMoney } from '../../utils/transformFunctions';

type ProductDataType = {
  productData: PropsCart
}

const Card: React.FC<ProductDataType> = ({ productData }) => {
  const navigation = useNavigation<StackNavigation>();
  return (


    <TouchableOpacity style={styles.containerCard} onPress={() => { navigation.navigate('Produto', productData); }} key={productData.id} testID={`imageProdutoButton${productData.id}`} >
      <Image source={{
        uri: productData.image,
      }} style={styles.produtoImage} testID={`imageProdudo${productData.id}`}></Image>
      <View style={styles.containerTextCard}>
        <Text style={styles.produtoTextTitle}>{productData.title}</Text>
        <Text style={styles.produtoTextPreco}>{transformMoney(productData.price)}</Text>
      </View>
    </TouchableOpacity>


  );
};

const styles = StyleSheet.create({

  containerCard: {
    display: 'flex',
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    width: 210,
    height: 290,
    flexDirection: 'column',
    background: '#c52222',
    borderRadius: 15,
    border: '1px solid #000',
    padding: 2,
    marginHorizontal: 10,

  },

  produtoImage: {
    width: 140,
    height: 120,
    borderRadius: 15
  },
  containerTextCard: {
    height: "40%",
    width: '100%',
    alignItems: 'center',
    display: 'flex',

  },
  produtoTextTitle: {
    fontSize: 16,
    fontWeight: '800',
    color: '#000'
  },
  produtoTextPreco: {
    fontSize: 14,
    fontWeight: '800',
    color: '#ed0e0e'
  }
});

export default Card