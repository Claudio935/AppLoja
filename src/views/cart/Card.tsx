import React, { useContext, } from 'react';
import {
  Image,
  StyleSheet,
  Text,
  View,
  Pressable,
  Alert
} from 'react-native';
import { CartContext } from '../../contexts/cartProvider/context';
import { type PropsCart } from '../../types/types';
import { IncrementProduct, decrementProduct, deleteProduct } from '../../contexts/cartProvider/actions';
import { transformMoney } from '../../utils/transformFunctions';


function Card(): JSX.Element {

  const { store: { cart }, dispatch } = useContext(CartContext);


  const handleIncrementProduct = (objectProduct: PropsCart): void => {

    if (typeof objectProduct?.id === "string" && objectProduct?.id.length === 0) {

      return
    };
    const newQuantidade = objectProduct.quantidade + 1
    const newObjectProduct = { ...objectProduct, quantidade: newQuantidade }
    IncrementProduct(dispatch, newObjectProduct)


  };
  const handleDecrementProduct = (objectProduct: PropsCart): void => {

    if (typeof objectProduct?.id === "string" && objectProduct?.id.length === 0) {
      return

    };
    if (objectProduct.quantidade === 0) {
      return
    }
    const newQuantidade = objectProduct.quantidade - 1

    const newObjectProduct = { ...objectProduct, quantidade: newQuantidade }
    decrementProduct(dispatch, newObjectProduct)


  };
  const handleDeleteProduct = (objectProduct: PropsCart): void => {
    if (typeof objectProduct?.id === "string" && objectProduct?.id.length === 0) {
      Alert.alert(
        "Alerta",
        "NÃ£o existe id para esse produto",
        [
          {
            text: "Cancelar",
            style: "cancel"
          },

        ]
      );

    };
    deleteProduct(dispatch, objectProduct)
  };

  return (
    <>
      {cart.map((item, index) => {
        return (
          <View style={styles.containerCarrinhoItem} key={index} >
            <Image source={{ uri: item.image }} style={{ width: '40%', height: '100%' }} />
            <View style={styles.containerCarrinhoTextButton}>
              <View style={styles.containerCarrinhoText}>
                <Text style={styles.titleProduto}>{item.title}</Text>
                <Text style={styles.quantidade}>Quantidade: {item.quantidade}</Text>
                <Text style={styles.price}>{transformMoney(item.price * item.quantidade)}</Text>
                <View style={styles.viewDecrementIncremment}>
                  <Pressable style={styles.buttonIncrement} onPress={() => { handleIncrementProduct(item); }} testID={`incrementTouch${index}`}>
                    <Text style={styles.textButton}>+</Text>
                  </Pressable>
                  <Pressable testID={`decrementTouch${index}`} style={styles.buttonDecrement} onPress={() => { handleDecrementProduct(item); }}>
                    <Text style={styles.textButton}>-</Text>
                  </Pressable>
                </View>
              </View>
              <Pressable testID={`retirarCarrinho${index}`} style={styles.button} onPress={() => { handleDeleteProduct(item); }} >
                <Text style={styles.textButton}>Retirar do Carrinho</Text>
              </Pressable>
            </View>
          </View>
        )
      })}
    </>
  );
};

const styles = StyleSheet.create({
  containerHeader: {
    marginBottom: 20
  },
  titleCarrinho: {
    textAlign: 'center',
    fontWeight: '900',
    color: '#000',
    fontSize: 30,
    marginBottom: 5
  },
  container: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column',
    width: '100%',
    padding: 20
  },
  containerCarrinhoItem: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    flexDirection: 'row',
    marginTop: 10,
    marginBottom: 10,
    height: 190,
  },
  containerCarrinhoText: {
    display: 'flex',
    justifyContent: 'space-around',
    alignItems: 'center',
    flexDirection: 'column',
    width: '90%',
    height: '70%'
  },
  containerCarrinhoTextButton: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column',
    width: '60%',
    height: '100%'
  },
  viewDecrementIncremment: {
    display: 'flex',
    justifyContent: 'space-evenly',
    flexDirection: 'row',
    paddingVertical: 15,
    width: 100
  },
  buttonIncrement: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: 35,
    width: 35,
    backgroundColor: '#1acd20',
    borderRadius: 100,
    padding: 7,
  },
  buttonDecrement: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: 35,
    width: 35,
    backgroundColor: '#fe0f0f',
    borderRadius: 100,
    padding: 7,
  },
  button: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: 35,
    backgroundColor: '#fe0f0f',
    borderRadius: 15,
    padding: 7,
  },
  textButton: {
    color: '#fff',
    fontWeight: '900'
  },
  titleProduto: {
    fontSize: 16,
    color: '#000',
    fontWeight: '900'
  },
  quantidade: {
    fontSize: 14,
    color: '#000',
    fontWeight: '700',
    paddingVertical: 15
  },
  price: {
    fontSize: 14,
    color: '#b71919',
    fontWeight: '900',
    padding: 5
  }
});


export default Card;