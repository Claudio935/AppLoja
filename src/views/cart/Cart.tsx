import { useNavigation } from '@react-navigation/native';
import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  Button,
} from 'react-native';
import { type StackNavigation } from '../../../App';
import Card from './Card';


function Cart(): JSX.Element {


  const navigation = useNavigation<StackNavigation>();


  return (
    <ScrollView>
      <View style={styles.container}>
        <View style={styles.containerHeader}>
          <Text style={styles.titleCarrinho}>Itens do carrinho</Text>
          <Button testID="voltarHomeButton" title='Adicionar novos itens' color={'#1acd20'} onPress={() => { navigation.navigate('Home'); }}></Button>
        </View>
        <Card />
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  containerHeader: {
    marginBottom: 20
  },
  titleCarrinho: {
    textAlign: 'center',
    fontWeight: '900',
    color: '#000',
    fontSize: 30,
    marginBottom: 5
  },
  container: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column',
    width: '100%',
    padding: 20
  },
  containerCarrinhoItem: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    flexDirection: 'row',
    marginTop: 10,
    marginBottom: 10,
    height: 190,
  },
  containerCarrinhoText: {
    display: 'flex',
    justifyContent: 'space-around',
    alignItems: 'center',
    flexDirection: 'column',
    width: '90%',
    height: '70%'
  },
  containerCarrinhoTextButton: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column',
    width: '60%',
    height: '100%'
  },
  viewDecrementIncremment: {
    display: 'flex',
    justifyContent: 'space-evenly',
    flexDirection: 'row',
    paddingVertical: 15,
    width: 100
  },
  buttonIncrement: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: 35,
    width: 35,
    backgroundColor: '#1acd20',
    borderRadius: 100,
    padding: 7,
  },
  buttonDecrement: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: 35,
    width: 35,
    backgroundColor: '#fe0f0f',
    borderRadius: 100,
    padding: 7,
  },
  button: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: 35,
    backgroundColor: '#fe0f0f',
    borderRadius: 15,
    padding: 7,
  },
  textButton: {
    color: '#fff',
    fontWeight: '900'
  },
  titleProduto: {
    fontSize: 16,
    color: '#000',
    fontWeight: '900'
  },
  quantidade: {
    fontSize: 14,
    color: '#000',
    fontWeight: '700',
    paddingVertical: 15
  },
  price: {
    fontSize: 14,
    color: '#b71919',
    fontWeight: '900',
    padding: 5
  }
});


export default Cart;