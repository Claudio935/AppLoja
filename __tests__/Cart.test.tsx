import React from 'react'
import { NavigationContainer } from "@react-navigation/native";
import { render, screen, fireEvent } from "@testing-library/react-native";
import "@testing-library/jest-dom";
import { Alert } from 'react-native';
import renderer from 'react-test-renderer';
import { CartContext } from '../src/contexts/cartProvider/context';
import { mockUseContext } from '../src/data/dadosMock/dadosMock';
import Cart from '../src/views/cart/Cart';
import { transformMoney } from '../src/utils/transformFunctions';
import Card from '../src/views/cart/Card';


describe("<Carrinho />", () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<CartContext.Provider value={mockUseContext}>
        <NavigationContainer>
          <Cart />
        </NavigationContainer>
      </CartContext.Provider >)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render the Cart page with products", async () => {


    const { debug } = render(
      <CartContext.Provider value={mockUseContext}>
        <Card />
      </CartContext.Provider >
    );

    debug()
    // verificando a existência do produto
    await screen.findByText("Retirar do Carrinho");
    const titleProduto = screen.getByText("testMasculino");
    const textQuantidade = screen.getByText("Quantidade: 1");
    const textPreco = screen.getByText(transformMoney(mockUseContext.store.cart[0].price));

    expect(titleProduto).toBeTruthy();
    expect(textQuantidade).toBeTruthy();
    expect(textPreco).toBeTruthy();


  });

  it("displays alert message when clicking the button", async () => {
    const alertMock = jest.spyOn(Alert, 'alert');


    const mockCarrinhoNotId = {
      store: {
        cart: [{
          title: 'testMasculino',
          image: './img/testMasculino',
          price: 1.5,
          description: 'testandoMasculino',
          id: "",
          category: "men's clothing",
          quantidade: 1,
        }],
      },
      dispatch: jest.fn()
    };


    const { getByTestId } = render(
      <CartContext.Provider value={mockCarrinhoNotId}>
        <NavigationContainer  >
          <Cart />
        </NavigationContainer>
      </CartContext.Provider>

    );
    // verificando a existência do produto

    const textRetirarCorrinho = await screen.findByText("Retirar do Carrinho");
    expect(textRetirarCorrinho).toBeTruthy()

    // ao pressionar o delete
    const retirarButton = getByTestId("retirarCarrinho0");

    fireEvent.press(retirarButton);

    expect(alertMock).toHaveBeenCalledWith("Alerta",
      "Não existe id para esse produto",
      [
        {
          text: "Cancelar",
          style: "cancel"
        },

      ]);


  });


})