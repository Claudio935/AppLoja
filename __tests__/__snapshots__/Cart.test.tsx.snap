// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carrinho /> renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "padding": 20,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "marginBottom": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 30,
              "fontWeight": "900",
              "marginBottom": 5,
              "textAlign": "center",
            }
          }
        >
          Itens do carrinho
        </Text>
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="voltarHomeButton"
        >
          <View
            style={
              [
                {},
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  {
                    "color": "#1acd20",
                  },
                ]
              }
            >
              Adicionar novos itens
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "height": 190,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Image
          source={
            {
              "uri": "./img/testMasculino",
            }
          }
          style={
            {
              "height": "100%",
              "width": "40%",
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "center",
              "width": "60%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "height": "70%",
                "justifyContent": "space-around",
                "width": "90%",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 16,
                  "fontWeight": "900",
                }
              }
            >
              testMasculino
            </Text>
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 14,
                  "fontWeight": "700",
                  "paddingVertical": 15,
                }
              }
            >
              Quantidade: 
              1
            </Text>
            <Text
              style={
                {
                  "color": "#b71919",
                  "fontSize": 14,
                  "fontWeight": "900",
                  "padding": 5,
                }
              }
            >
              R$Â 1,50
            </Text>
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                  "paddingVertical": 15,
                  "width": 100,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#1acd20",
                    "borderRadius": 100,
                    "display": "flex",
                    "height": 35,
                    "justifyContent": "center",
                    "padding": 7,
                    "width": 35,
                  }
                }
                testID="incrementTouch0"
              >
                <Text
                  style={
                    {
                      "color": "#fff",
                      "fontWeight": "900",
                    }
                  }
                >
                  +
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#fe0f0f",
                    "borderRadius": 100,
                    "display": "flex",
                    "height": 35,
                    "justifyContent": "center",
                    "padding": 7,
                    "width": 35,
                  }
                }
                testID="decrementTouch0"
              >
                <Text
                  style={
                    {
                      "color": "#fff",
                      "fontWeight": "900",
                    }
                  }
                >
                  -
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fe0f0f",
                "borderRadius": 15,
                "display": "flex",
                "height": 35,
                "justifyContent": "center",
                "padding": 7,
              }
            }
            testID="retirarCarrinho0"
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontWeight": "900",
                }
              }
            >
              Retirar do Carrinho
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
